source/processes/fs.cpp

```diff
@@ -2747,10 +2782,72 @@ std::tuple<OS::Result, IPC::MappedBuffer>
 FakeFS::HandleCreateExtSaveData(FakeThread& thread, ProcessId, const Platform::FS::ExtSaveDataInfo& info,
                                 uint32_t max_directories, uint32_t max_files, uint64_t, uint32_t,
                                 IPC::MappedBuffer input_smdh) {
-    auto result = ArchiveExtSaveData::CreateSaveData(thread, *this, info, max_directories, max_files, true);
+    auto result = ArchiveExtSaveData::CreateSaveData(thread, *this, info, max_directories, max_files, input_smdh, true);
     return std::make_tuple(result, input_smdh);
 }
 
+std::tuple<OS::Result, uint32_t, IPC::MappedBuffer>
+FakeFS::HandleEnumerateExtSaveData( FakeThread& thread, ProcessId, uint32_t, uint32_t media_type,
+                                    uint32_t id_entry_size, uint32_t is_shared, IPC::MappedBuffer output_ids) {
+    // Game carts cannot have extdata
+    if (media_type == 2)
+        return std::make_tuple(0xE0C046FA, 0, output_ids);
+
+    if (id_entry_size != 4 && id_entry_size != 8 || output_ids.size % id_entry_size)
```

Comment:
Combining && and || without parenthesis is error-prone. I'm guessing the intent here is `(id_entry_size != 4 && id_entry_size != 8) || output_ids.size % id_entry_size`, so let's be explicit about it.